src/backend/metal/BlendStateMTL.mm

```diff
@@ -0,0 +1,104 @@
+
+namespace backend {
+namespace metal {
+
+    namespace {
+
+        MTLBlendFactor MetalBlendFactor(nxt::BlendFactor factor, bool alpha) {
+            switch(factor) {
+                case nxt::BlendFactor::Zero:
+                    return MTLBlendFactorZero;
+                case nxt::BlendFactor::One:
+                    return MTLBlendFactorOne;
+                case nxt::BlendFactor::SrcColor:
+                    return MTLBlendFactorSourceColor;
+                case nxt::BlendFactor::OneMinusSrcColor:
+                    return MTLBlendFactorOneMinusSourceColor;
+                case nxt::BlendFactor::SrcAlpha:
+                    return MTLBlendFactorSourceAlpha;
+                case nxt::BlendFactor::OneMinusSrcAlpha:
+                    return MTLBlendFactorOneMinusSourceAlpha;
+                case nxt::BlendFactor::DstColor:
+                    return MTLBlendFactorDestinationColor;
+                case nxt::BlendFactor::OneMinusDstColor:
+                    return MTLBlendFactorOneMinusDestinationColor;
+                case nxt::BlendFactor::DstAlpha:
+                    return MTLBlendFactorDestinationAlpha;
+                case nxt::BlendFactor::OneMinusDstAlpha:
+                    return MTLBlendFactorOneMinusDestinationAlpha;
+                case nxt::BlendFactor::SrcAlphaSaturated:
+                    return MTLBlendFactorSourceAlphaSaturated;
+                case nxt::BlendFactor::BlendColor:
+                    return alpha ? MTLBlendFactorBlendAlpha : MTLBlendFactorBlendColor;
+                case nxt::BlendFactor::OneMinusBlendColor:
+                    return alpha ? MTLBlendFactorOneMinusBlendAlpha : MTLBlendFactorOneMinusBlendColor;
+                case nxt::BlendFactor::Src1Color:
+                    return MTLBlendFactorSource1Color;
+                case nxt::BlendFactor::OneMinusSrc1Color:
+                    return MTLBlendFactorOneMinusSource1Color;
+                case nxt::BlendFactor::Src1Alpha:
+                    return MTLBlendFactorSource1Alpha;
+                case nxt::BlendFactor::OneMinusSrc1Alpha:
+                    return MTLBlendFactorOneMinusSource1Alpha;
+            }
+        }
+
+        MTLBlendOperation MetalBlendOperation(nxt::BlendOperation operation) {
+            switch(operation) {
+                case nxt::BlendOperation::Add:
+                    return MTLBlendOperationAdd;
+                case nxt::BlendOperation::Subtract:
+                    return MTLBlendOperationSubtract;
+                case nxt::BlendOperation::ReverseSubtract:
+                    return MTLBlendOperationReverseSubtract;
+                case nxt::BlendOperation::Min:
+                    return MTLBlendOperationMin;
+                case nxt::BlendOperation::Max:
+                    return MTLBlendOperationMax;
+            }
+        }
+
+        MTLColorWriteMask MetalColorWriteMask(nxt::ColorWriteMask colorWriteMask) {
+            return (
```

Comment:
nit: Metal uses an inverse order compared to NXT so we can't do the same thing as for D3D12